### PR TITLE
FQM-297 Add service ignore list in case not all CDS services provided are intended to be CRD-conformant

### PR DESCRIPTION
**Pull request by arscan**
# Summary

In the case that the server provides CDS services that are outside the scope of CRD, the tester can input a comma-separated list of services to ignore.   It ignores them in discovery validation, and then does not persist them for later tests (thus effectively ignoring them for all tests).  This is included so that we do not over-constrain CDS servers in a way that prevents CRD and non-CRD services from being provided in one endpoint.

Screenshot 2026-05-06 at 2 08 48 PM

# Testing Guidance

Run the Discover test against the client tests preset.  By default, the discovery test should function the same.  If you add `appointment-book-service` an 'info' message will appear in the log stating that it is being ignored, and the associated hook will be gone from the OUTPUT variable.  There is no way to have the mock server advertise an additional non-CRD service.  I do not think we need to have our simulated server have this level of functionality as it is not mentioned in the spec.

### CODE REVIEW

File: lib/davinci_crd_test_kit/server/v2.2.1/discovery/discovery_services_validation_test.rb [hunk 1]
```diff
@@ -64,7 +85,7 @@ def required_fields
                order_select_service_ids: service_hooks_to_ids['order-select']&.join(', '),
                order_sign_service_ids: service_hooks_to_ids['order-sign']&.join(', ')
 
-        services.each do |service|
+        services_for_crd_validation.each do |service|
```
Comment (Jammjammjamm) [inline lib/davinci_crd_test_kit/server/v2.2.1/discovery/discovery_services_validation_test.rb:91]: I think we need to skip if this is blank. If you ignore all service ids, the test still passes.

Image

Image

**Review by Jammjammjamm — APPROVED** [review]

File: lib/davinci_crd_test_kit/server/v2.2.1/discovery/discovery_services_validation_test.rb [hunk 2]
```diff
@@ -64,7 +88,7 @@ def required_fields
                order_select_service_ids: service_hooks_to_ids['order-select']&.join(', '),
                order_sign_service_ids: service_hooks_to_ids['order-sign']&.join(', ')
```
Comment (tstrass) [inline lib/davinci_crd_test_kit/server/v2.2.1/discovery/discovery_services_validation_test.rb]: Contrived edge case, but as it stands if you run discovery once and then add the rest of the services to the ignore list and rerun discovery, the newly ignored service IDs will still be seen and used by subsequent tests. So I think it would be better to put the `skip_if` after outputting the service IDs.

**Review by tstrass — CHANGES_REQUESTED** [review]

File: lib/davinci_crd_test_kit/server/v2.2.1/discovery/discovery_services_validation_test.rb (hunk 2 above)
Comment (arscan) [reply lib/davinci_crd_test_kit/server/v2.2.1/discovery/discovery_services_validation_test.rb]: Hmm ok. I was following the lead of the earlier skip_if to skip early, which exhibits a similar problem but perhaps its more likely that someone will accidentally ignore all services via input after an initial run than break service discovery response after an initial run in their implementation?

Comment (tstrass) [reply lib/davinci_crd_test_kit/server/v2.2.1/discovery/discovery_services_validation_test.rb]: Ah I see. I think it might be best to have both of those skip cases affect the test session state. But like I said it's a pretty contrived scenario, so I'm fine with leaving this with behavior consistent with the other skip.

**Review by tstrass — APPROVED** [review]

**Review by tstrass — APPROVED** [review]
